### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by Vuwij**
https://packages.ubuntu.com/xenial/docker.io
https://packages.ubuntu.com/xenial/docker-compose

https://apps.fedoraproject.org/packages/docker
https://apps.fedoraproject.org/packages/docker-compose

https://packages.debian.org/buster/docker.io
https://packages.debian.org/stretch/docker-compose

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -602,6 +602,17 @@ dnsmasq:
   fedora: [dnsmasq]
   gentoo: [net-dns/dnsmasq]
   ubuntu: [dnsmasq]
+docker:
```
Comment (nuclearsandwich) [inline rosdep/base.yaml]: If you don't need the System Tray docklet which is installed via the `docker` package on Debian and Ubuntu this key can be removed.

File: rosdep/base.yaml [hunk 2]
```diff
@@ -602,6 +602,17 @@ dnsmasq:
...
+  debian: [docker]
+  fedora: [docker]
```
Comment (nuclearsandwich) [inline rosdep/base.yaml]: The `docker` packages in Debian and Ubuntu refer to something entirely different from the `docker` package in Fedora. Fedora's `docker` package belongs with the `docker.io` keys for Debian and Ubuntu below.

**Review by nuclearsandwich — CHANGES_REQUESTED** [review]

**Comment by nuclearsandwich** [comment]
@Vuwij are you intentionally adding keys for the `docker` package in Stretch and Xenial which is described as a "System tray for KDE3/GNOME2 docklet applications" and is not related to the Moby/Docker project?

File: rosdep/base.yaml [hunk 3]
```diff
@@ -602,6 +602,17 @@ dnsmasq:
...
+  ubuntu: [docker]
+docker-compose:
+  debian: [docker-compose]
+  fedora: [docker-compose]
+  ubuntu: [docker-compose]
+docker.io:
+  debian: [docker.io]
```
Comment (tfoote) [inline rosdep/base.yaml]: From the link it appears that docker.io is only available on buster and newer. We should explicitly set None for the other currently supported distros so people don't get surprised when it doesn't work. And use a wildcard for buster and newer.

File: rosdep/base.yaml (hunk 1 above)
Comment (tfoote) [reply rosdep/base.yaml]: This does look like it's unrelated.

**Review by tfoote — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml (hunk 3 above)
Comment (Vuwij) [reply rosdep/base.yaml]: Is this correct now?

**Comment by Vuwij** [comment]
Is it correct now?

**Comment by Vuwij** [comment]
@tfoote @nuclearsandwich should it be ok now?

**Review by Vuwij — COMMENTED** [review]
Ok

**Review by Vuwij — COMMENTED** [review]
Ok

**Review by nuclearsandwich — APPROVED** [review]

**Comment by Vuwij** [comment]
bump

**Review by tfoote — APPROVED** [review]

**Pull request merged**